crates/astria-core/src/protocol/transaction/v1alpha1/action.rs

```diff
@@ -442,14 +442,20 @@ impl SequenceAction {
             fee_asset,
         })
     }
+
+    /// Returns action type as `&str` for fee event logging
+    #[must_use]
+    pub fn get_action_type(&self) -> &str {
+        "Sequence"
```

Comment:
The actions are refinement types of types generated protobuf code. Please use https://docs.rs/prost/latest/prost/trait.Name.html instead (preferrably `Name::full_name()`, even though that allocates).

We can achieve this by updating the `astria_core::Protobuf` trait: it already has an associated type `Raw`. We should update it like so:

```rust
trait Protobuf {
    type Raw: prost::Name;

    fn full_name() -> String {
        Self::raw::full_name()
    }
}
```

Reply:
I may be mistaken, but it looks like `Action` and its variants don't implement the `Protobuf` trait, so even with the changes to Protobuf, the `full_name()` function is not inherited. It does, however, look like the generated action types already implement `prost::Name::full_name()`, so I have changed `get_action_type()` to look like so:
```rust
/// Returns full name of action.
#[must_use]
pub fn full_name() -> String {
    raw::TransferAction::full_name()
}
```
Let me know if you think this is a satisfactory solution, or if you had something else in mind!

Reply:
IMO just impl the trait for the types. We should probably just go ahead and do that for all the types anyway.

Reply:
I was going to ask, seemed like they all should probably implement the trait. Should I go ahead and do the implementation for all in this PR? Or would it be best to save that for a new ticket?

Reply:
Fine to do it for the subset of actions that you need right now. Can you go ahead and create a ticket for the others?

Reply:
I'll do that. Thanks for the help!